e2e/tests/eth_integration.rs

```diff
@@ -193,7 +193,7 @@ async fn eth_integration(web3: Web3) {
             web3: web3.clone(),
         }),
     );
-    let solver = solver::solver::naive_solver();
+    let solver = solver::solver::naive_solver(solver);
```

Comment:
Could we maybe call the internal one something different like `solver_wallet` or `unlocked_solver_wallet`?